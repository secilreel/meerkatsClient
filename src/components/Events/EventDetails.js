import React,{Component} from 'react';
import {Link} from 'react-router-dom';
import calendar from '../../Images/calendar.svg';
import clock from '../../Images/clock.svg';
import map from '../../Images/map.svg';

export default class EventDetails extends Component {
    constructor(props) {
        super(props);
    this.state={
        event: this.props.event,
        participants: this.props.participants
        }
    }

    handleClickCloseButton=()=>{
        this.props.history.goBack()
    }

    render(){
        console.log(this.props.participants)
        const date= new Date (this.props.event.meeting_day).toDateString();
    return (
        <section className="event container">
            <h2>{this.props.event.title}</h2>
            <div className="button-box">
                <button type="button" onClick={this.handleClickCloseButton}>Close</button>
            </div>
            <div className="event-box">
                    <img src={this.props.event.image} className="event-owner logo" alt="headshot of the event planner's account" />
                    <div className="event-details">
                        <p>{this.props.event.details}</p>
                        <div className="event-icon-box">
                        <img src={map} alt="clock icon" className="event-icons"/>
                        <div>{this.props.event.place}</div>
                        </div>
                        <div className="event-icon-box">
                        <img src={calendar} alt="calendar icon" className="event-icons" />
                        <p>{date}</p>
                        </div>
                        <div className="event-icon-box">
                        <img src={clock} alt="clock icon" className="event-icons"/>
                        <p>{this.props.event.meeting_time}</p>
                        </div>
                        <div className="attendees">
                            <ul>
                            {this.props.participants.map(participant =>
                            <Link to={`/status/${this.props.eventId}`} key={participant.id}>
                            <li className="participant-box"
                            >
                            <img src={participant.image} className="attandees logo" alt="headshot of the attandee" />
                            {participant.user_name}: {participant.attending}
                            </li>
                            <br />
                            </Link>
                            )
                            }
                    
                            </ul>
                        </div>
                    </div>
                </div>
                <div className="button-box">
                    <button type="button" className="js-remove-event-button">Remove Event</button>
                </div>        
            </section>
    );
}
}