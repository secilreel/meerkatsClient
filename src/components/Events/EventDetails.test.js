import React from 'react';
import ReactDOM from 'react-dom';
import EventDetails from './Event';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

configure({ adapter: new Adapter() });

describe('<Event />', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <EventDetails />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
    <MemoryRouter>
    <EventDetails 
    title="test Event"
    date="Fri March 30, 2019"
    time="10:30:00"
    place="Boston"
        />
    </MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });

it('renders empty given no selection', () => {
    const wrapper = shallow(<EventDetails />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

// it('Should call props.onClick() when the Delete button is clicked', () => {
//     const clickHandler = () => {console.log('button clicked')};
//     const wrapper = shallow(<EventDetails onClick={clickHandler}/>);
//     wrapper.find('button').at(0).simulate('click');
//     expect(toJson(wrapper)).toMatchSnapshot();
// });
})
