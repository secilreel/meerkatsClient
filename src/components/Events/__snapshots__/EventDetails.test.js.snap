// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> renders empty given no selection 1`] = `
<Link
  replace={false}
  to="/events/undefined"
>
  <section
    className="event container"
  >
    <h2 />
    <div
      className="event-box"
    >
      <img
        alt="headshot of the event planner"
        className="event-owner logo"
      />
      <div
        className="event-summary"
      >
        <div
          className="event-icon-box"
        >
          <img
            alt="calendar icon"
            className="event-icons"
            src="calendar.svg"
          />
          <div>
            Invalid Date
          </div>
        </div>
        <div
          className="event-icon-box"
        >
          <img
            alt="clock icon"
            className="event-icons"
            src="clock.svg"
          />
          <div>
            undefined
          </div>
        </div>
        <div
          className="event-icon-box"
        >
          <img
            alt="map icon"
            className="event-icons"
            src="map.svg"
          />
          <div>
            undefined
          </div>
        </div>
      </div>
    </div>
    <div
      className="button-box"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Remove Event
      </button>
    </div>
  </section>
</Link>
`;

exports[`<Event /> renders the UI as expected 1`] = `
<a
  href="/events/undefined"
  onClick={[Function]}
>
  <section
    className="event container"
  >
    <h2>
      test Event
    </h2>
    <div
      className="event-box"
    >
      <img
        alt="headshot of the event planner"
        className="event-owner logo"
      />
      <div
        className="event-summary"
      >
        <div
          className="event-icon-box"
        >
          <img
            alt="calendar icon"
            className="event-icons"
            src="calendar.svg"
          />
          <div>
            Invalid Date
          </div>
        </div>
        <div
          className="event-icon-box"
        >
          <img
            alt="clock icon"
            className="event-icons"
            src="clock.svg"
          />
          <div>
            10:30:00
          </div>
        </div>
        <div
          className="event-icon-box"
        >
          <img
            alt="map icon"
            className="event-icons"
            src="map.svg"
          />
          <div>
            Boston
          </div>
        </div>
      </div>
    </div>
    <div
      className="button-box"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Remove Event
      </button>
    </div>
  </section>
</a>
`;
